app/models/case_log.rb

```diff
@@ -1,5 +1,41 @@
-class CaseLog < ApplicationRecord
+class CaseLogValidator < ActiveModel::Validator
+
+  # Methods need to be named 'validate_' followed by field name
+  # this is how the metaprogramming of the method name being 
+  # call in the validate method works.
+
+  def validate_tenant_code(record)
+    if record.tenant_code.blank?
+      record.errors.add :tenant_code, "Tenant code can't be blank"
+    end
+  end
+
+  def validate_tenant_age(record)
+    if record.tenant_age.blank?
+      record.errors.add :tenant_age, "Tenant age can't be blank"
+    elsif !(record.tenant_age.to_s =~ /^[1-9][0-9]?$|^100$/)
+      record.errors.add :tenant_age, "Tenant age must be between 0 and 100"
+    end
+  end
+
+  def validate(record)
+    question_to_validate = options[:previous_page]
```

Comment:
I think this might have to be a loop? If you had multiple things on a page to validate?